RELEASE.md

```diff
@@ -10,20 +10,15 @@
 # Upcoming Release 0.18.0
 
```

Comment:
I'm also thinking it would be great to add in a TL;DR for this release just above "Major features and improvements". It could read something like: "TL;DR Kedro 0.18.0 strives to reduce the complexity of the project template and get us closer to a stable release of the framework. We've also introduced our full [micro-packaging workflow](https://kedro.readthedocs.io/en/stable/06_nodes_and_pipelines/04_micro_packaging.html), which allows you to import packages, utilities functions and existing pipelines into your Kedro project. The release also comes with long-awaited Python 3.9 and 3.10 support." 

But I'll also look to @idanov to add more comment on the release too.